pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -40,6 +40,19 @@ type EC2Filter struct {
 	Values []string `json:"values"`
 }
 
+// K8SRole is role of the service in Kubernetes.
+// Currently the only supported role is "Node".
+// +kubebuilder:validation:Enum=Node;node
+type K8SRole string
+
+// K8SSelectorConfig is Kubernetes Selector Config
+type K8SSelectorConfig struct {
+	// +kubebuilder:validation:Required
+	Role  K8SRole `json:"role,omitempty"`
```

Comment:
we need to remove the `omitempty` part to make it really required.
```suggestion
	Role  K8SRole `json:"role"`
```

Reply:
aah my bad I forgot to remove it here, updated